File: frontend/src/api/feedback.ts
import api from "./client";

export interface Feedback {
  id: number;
  lesson_id: number;
  student_id: number;
  rating: number;
  comment?: string | null;
  is_hidden: boolean;
  created_at: string;
  student_name?: string | null;
  course_name?: string | null;
  lesson_topic?: string | null;
  lesson_date?: string | null;
}

export interface FeedbackCreate {
  lesson_id: number;
  student_id?: number;
  rating: number;
  comment?: string | null;
  is_hidden?: boolean;
}

export interface FeedbackUpdate {
  rating?: number;
  comment?: string | null;
  is_hidden?: boolean;
}

export interface FeedbackQueryParams {
  include_hidden?: boolean;
}

export async function getFeedback(params?: FeedbackQueryParams): Promise<Feedback[]> {
  const response = await api.get("/feedback", { params });
  return response.data as Feedback[];
}

export async function createFeedbackItem(payload: FeedbackCreate): Promise<Feedback> {
  const response = await api.post("/feedback", payload);
  return response.data as Feedback;
}

export async function updateFeedbackItem(feedbackId: number, payload: FeedbackUpdate): Promise<Feedback> {
  const response = await api.patch(`/feedback/${feedbackId}`, payload);
  return response.data as Feedback;
}

File: backend/app/schemas/feedback.py
from datetime import datetime, date
from typing import Optional

from pydantic import BaseModel, confloat


class FeedbackBase(BaseModel):
    lesson_id: int
    student_id: int
    rating: confloat(ge=0, le=5)
    comment: Optional[str] = None
    is_hidden: bool = False


class FeedbackCreate(FeedbackBase):
    pass


class FeedbackUpdate(BaseModel):
    rating: Optional[confloat(ge=0, le=5)] = None
    comment: Optional[str] = None
    is_hidden: Optional[bool] = None


class FeedbackRead(FeedbackBase):
    id: int
    created_at: datetime
    student_name: Optional[str] = None
    course_name: Optional[str] = None
    lesson_topic: Optional[str] = None
    lesson_date: Optional[date] = None

    class Config:
        from_attributes = True
